ts/Stock/Indicators/SMA/SMAIndicator.ts

```diff
@@ -397,6 +394,19 @@ class SMAIndicator extends LineSeries {
             max,
             i;
 
+        // Updating an indicator with redraw=false may destroy data.
+        // If there will be a following update for the parent series,
+        // we will try to access Series object without any properties
+        // (except for prototyped ones). This is what happens
+        // for example when using Axis.setDataGrouping(). See #16670
+        processedData = indicator.linkedParent.options ?
```

Comment:
I'm curious if there's a better way to check whether the series is being removed/updated. I decided to use `options`, because when there are `options` there must be a series too.

Reply:
> I'm curious if there's a better way to check whether the series is being removed/updated

Usually we use `series.isDirtyData` for that purpose. Did you try that? Or maybe it doesn't cover this situation?

Reply:
Thanks! Let me check..
Checked `series.isDirtyData`: it seems it can be undefined (and be valid), so we can accidentally skip processing the data, not good. Double-checked alternatives (according to TS and `series.init`): `chart`, `userOptions`, `data`, `eventsToUnbind`, `xAxis`, `yAxis`, `index`, `linkedSeries`, `tooltipOptions`, `points` (this one is definitively incorrect, DG can nullify this property), `zones` and some others.

I think, we used to have a property like `isRemoving`, but I don't see it anymore.

Reply:
Have you considered the possibility that we may be forgetting to set `isDirtyData` in places where it should be set? The purpose of this flag is to set it whenever we are doing create, update or delete operations on the series data. So if we have a scenario where we modify the data and `isDirtyData` is not `true`, it should be considered a bug.

Reply:
I have not. Good point, let me check that :)

Reply:
Okay, here is the process:
1. Click the RangeSelector's button which has `dataGrouping` set
2. Delegate to `Axis.setDataGrouping`, which updates each series with new `dataGrouping` config:
    - update indicator series with new DG
        - indicator is removed, the connection with `linkedParent` is broken (which is correct, `update()` might change the parent etc.)
        - `afterLinkSeries` event is fired, but indicator was not initialized, so far so good
        - indicator is added back with new options, awaits `afterLinkSeries` event, to connect to the parent
    - update Navigator with new DG
        - `series.update()` calls `series.remove()` which calls `chart.linkSeries()`
        - indicator's parent series is still available, so linking succeeds
    - update the main series with new DG
        - `series.update()` calls `series.remove()` which calls `chart.linkSeries()`
        - because `linkSeries()` is inside `series.remove()`, the series is not completely removed. It's just an object with methods and properties from Prototype. Indicator still holds the reference to this series. Tries to access `series.something[0]` and the errors occures.

I think, there's no spot where `isDirtyData` should be set.

Extra:
In fact, I just realized that `linkSeries` clears `series.linkedSeries` array, but does not clear `series.linkedParent`. Just tried that solution [demo](https://jsfiddle.net/BlackLabel/pdyx0935/) - now it throws our error, that parent series was not found. This is correct because the parent series was removed.

Reply:
Ok, thanks for the explanation!